fathom/src/surface/elaboration/unification.rs

```diff
@@ -212,15 +216,20 @@ impl<'arena, 'env> Context<'arena, 'env> {
                 if prim0 == prim1 =>
             {
                 self.unify_spines(spine0, spine1)
+                    .map_err(map_spine_mismatch(value0, value1))
             }
             (
                 Value::Stuck(Head::RigidVar(var0), spine0),
                 Value::Stuck(Head::RigidVar(var1), spine1),
-            ) if var0 == var1 => self.unify_spines(spine0, spine1),
+            ) if var0 == var1 => self
+                .unify_spines(spine0, spine1)
+                .map_err(map_spine_mismatch(value0, value1)),
```

Comment:
What was the reasoning behind mapping to a `Mismatch` here?

Reply:
That was the original behaviour in `unify_spines`:

```rust
        if spine0.len() != spine1.len() {
            return Err(Error::Mismatch);
```